src/Checkbox/index.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react'
+import { Checkbox as AntCheckbox, CheckboxProps as AntCheckboxProps } from 'antd'
+import 'antd/lib/checkbox/style/index.css'
+import './style.css'
+
+export type CheckboxProps = AntCheckboxProps
+const Checkbox: React.FC<CheckboxProps> = props => {
+  const { indeterminate, checked, ...restProps } = props
+
+  const [checkIndeterminate, setIndeterminate] = React.useState(indeterminate)
+  const [checkAll, setCheckAll] = React.useState(checked)
+
+  const onCheckAllChange = (e: any) => {
+    const option = !checkIndeterminate
+    setIndeterminate(option)
+    setCheckAll(e.target.checked)
+  }
```

Comment:
This logic's only purpose seems to be related to showing the component inside the story book. In this case it should be probably moved to the story book logic.

Reply:
Storybook Index stories does not support hooks, it will shows "Invalid hook call. Hooks can only be called inside of the body of a function component." if implement inside the index stories.

Reply:
@Kelvin-GohKengBoon You can create and export a wrapper component inside `index.stories.tsx` that contains the state management logic and any external logic required for storybook.

Reply:
Resolved.